### PR TITLE
Protect dossier participations endpoints with modify permission.

### PR DESCRIPTION
**Pull request by tinagerber**
Participations should not be able to be modified if a dossier cannot be modified.

Jira: https://4teamwork.atlassian.net/browse/NE-219

## Checklist (Must have)

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

### CODE REVIEW

File: opengever/api/tests/test_dossier_participations.py [hunk 1]
```diff
@@ -254,6 +254,19 @@ def test_post_participation_with_existing_participant_raises_bad_request(self, b
              "type": "BadRequest"},
             browser.json)
 
+    @browsing
+    def test_post_participation_for_expired_dossier_raises_unauthorized(self, browser):
```
Comment (njohner) [inline opengever/api/tests/test_dossier_participations.py:258]: Whether it's expired or not makes no difference here. What you want is test for a resolved dossier. the `expired_dossier` is the only resolved dossier in the fixture though. But for clarity I would change the test name. Then you can either leave the rest as is, or use the `resolvable_dossier` and resolve it in the test.

File: opengever/api/tests/test_dossier_participations.py [hunk 2]
```diff
@@ -425,6 +451,18 @@ def test_delete_participations_with_invalid_participant_id_raises_bad_request(se
         self.assertEqual({"message": "chaosqueen is not a valid id",
                           "type": "BadRequest"}, browser.json)
 
+    @browsing
+    def test_delete_participation_for_offered_dossier_raises_unauthorized(self, browser):
```
Comment (njohner) [inline opengever/api/tests/test_dossier_participations.py:455]: Again whether the dossier is offered or not does not make a difference here...

**Review by njohner — COMMENTED** [review]
does this work correctly in the classic UI already? Is there a test?

Otherwise LGTM.

**Comment by tinagerber** [comment]
@njohner Why should I only test resolved dossiers? It's more about testing that you can't modify participations if the dossier can't be modified. This is also true for other states. 

It already works in the old UI.

**Comment by njohner** [comment]
Expired is not a workflow state. An expired dossier is in the resolved state. So for that test I'd really rename it. IMO it's clearer that you are testing the "Modify" permission when you look at a resolved dossier. Otherwise I just wonder why the dossier should be expired in order for the POST to raise unauthorized.

If you want to test `offered`, sure why not. From a workflow point of view, a dossier can be edited until it is resolved or inactivated. Offering and then archiving it does not change the state of the dossier much and is not much used by any clients yet.

**Comment by tinagerber** [comment]
Ah I didn't see that the ones in the fixtures are almost all in the "resolved" state, as there are definitely other states in the workflow. Then I will just test inactive and resolved dossiers.

**Review by njohner — APPROVED** [review]
LGTM

**Pull request merged**